src/server/qgswmsserver.cpp

```diff
@@ -1446,10 +1434,16 @@ QImage* QgsWmsServer::getMap( HitTest* hitTest )
   applyOpacities( layersList, bkVectorRenderers, bkRasterRenderers, labelTransparencies, labelBufferTransparencies );
 
   if ( hitTest )
-    runHitTest( &thePainter, *hitTest );
+    runHitTest( mapSettings, &thePainter, *hitTest );
   else
   {
-    mMapRenderer->render( &thePainter );
+    QgsMapRendererCustomPainterJob renderJob( mapSettings, &thePainter );
+#ifdef HAVE_SERVER_PYTHON_PLUGINS
+    renderJob.setFeatureFilterProvider( mAccessControl );
+#endif
+    renderJob.renderSynchronously();
+    //renderJob.start();
+    //renderJob.waitForFinished();
   }
```

Comment:
@pblottiere just guessing: maybe you miss an event loop? (A local event loop in the test might help if that's the case, connect to https://qgis.org/api/classQgsMapRendererJob.html#a8dbfbe2cc381f0eb6e721733838d3baf)